### DISCUSSION TITLE
Symfony Messenger custom headers

### DISCUSSION THREAD

**Discussion by samuel4x4** [symfony/symfony#58254] [Q&A]
Hello,

I need to attach a custom header like "x-message-deduplication" to some messages I send to RabbitMQ thru Symfony Messenger. Can anybody give me a solution for this please? 

If I create a custom Stamp and it generates me a header like this:

X-Message-Stamp-App\Messenger\DeduplicationStamp:	[{"uniqueHash":"dewf8347wfh84fhcre87fh"}]

but I need the header like this:

x-deduplication-header: dewf8347wfh84fhcre87fh

How can I accomplish this? Thanks!

**Comment by darton1** [comment]
```php
$bus->dispatch(new Message(), [
    AmqpStamp::createWithAttributes([
        'headers' => [
            'x-deduplication-header' => 'dewf8347wfh84fhcre87fh'
        ]
    ])
]);
```
